### PR TITLE
LogOutputStream.create(LineConsumer)

### PR DESCRIPTION
**Pull request by panchenko**
So instead of creating an anonymous class it can be one-liner with lambda expression.

### CODE REVIEW

**Comment by panchenko** [comment]
If we are going to proceed with this, then I can also add an example to the readme file.

File: src/main/java/org/zeroturnaround/exec/stream/LogOutputStream.java [hunk 1]
```diff
@@ -159,4 +158,27 @@ protected void processBuffer() {
    */
   protected abstract void processLine(String line);
 
+  /**
+   * This is equivalent to {@code java.util.function.Consumer} while staying compatible with
+   * earlier java version.
+   */
```
Comment (nemecec) [inline src/main/java/org/zeroturnaround/exec/stream/LogOutputStream.java:164]:
```suggestion
  /**
   * This is equivalent to {@code java.util.function.Consumer} while staying compatible with
   * Java versions earlier than 8.
   */
```

File: src/main/java/org/zeroturnaround/exec/stream/LogOutputStream.java [hunk 2]
```diff
@@ -159,4 +158,27 @@ protected void processBuffer() {
...
+  public interface LineConsumer {
+    void accept(String line);
+  }
+
+  /**
+   * Creates LogOutputStream passing each line to the specified consumer.
+   * This method is supposed to be used in java 8+, providing consumer as a lambda expression.
+   */
```
Comment (nemecec) [inline src/main/java/org/zeroturnaround/exec/stream/LogOutputStream.java:172]:
```suggestion
  /**
  * Factory method to create a <code>LogOutputStream</code> that passes each line to the specified consumer.
  * <p>Mostly useful with Java 8+, so the consumer can be passed as a lambda expression.</p>
  *
  * @param consumer the consumer to consume the log lines
   * @return the created <code>LogOutputStream</code>, passing each line to the specified consumer.
   */
```

**Review by nemecec — CHANGES_REQUESTED** [review]
Cool idea, thanks! I think the JavaDoc could be better (I added some comments with suggestions, feel free to use those as inspiration).
Also, a unit test and example usage in README would be much appreciated.

File: README.md [hunk 3]
```diff
@@ -155,7 +155,17 @@ new ProcessExecutor().command("java", "-version")
 
 <hr/>
 
-* Handling output line-by-line while process is running
+* Handling output line-by-line while process is running (java 8+)
+
+```java
+new ProcessExecutor().command("java", "-version")
+    .redirectOutput(LogOutputStream.create(line -> ...))
```
Comment (panchenko) [inline README.md:162]: Theoretically, there could be also a new method here `.processOutput(line -> ...)`, WDYT?

**Comment by aalmiray** [comment]
Bytecode compatibility is set to Java 6 as a minimum which will prevent the use of lambdas.
I'd suggest bumping to 7 as a minimum as JDK 17 (see PR for build) fails if 6 is chosen.
Another option would be to bump to Java 8 as a minimum, in which case lambdas may be used.

Bear in mind that bumping Java baseline constitutes a breaking change and thus the library should increase its major version number according to semver.

**Comment by panchenko** [comment]
@nemecec Could you lease take a look? Thanks!

**Comment by panchenko** [comment]
Is this project maintained?

**Comment by nemecec** [comment]
> Is this project maintained?

Sorry, I totally forgot about this PR. Looks good, I'll merge it in.

**Comment by nemecec** [comment]
Thanks for the PR!

**Pull request merged**